### PR TITLE
posix: relocate `uname` cmd from sample

### PR DESCRIPTION
**Pull request by ycsin**
Relocate the `uname` cmd from the uname sample and updated the sample to use only POSIX headers & APIs. The sample can now be compiled on Linux:

```
# Go to the sample dir
cd ${ZEPHYR_BASE}/samples/posix/uname

# Compile the sample
make -f Makefile.host

# Run the binary
./build/uname

sysname[65]: Linux
nodename[65]: LAPTOP-YC
release[65]: 5.10.16.3-microsoft-standard-WSL2
version[65]: https://github.com/ycsin/zephyr/pull/1 SMP Fri Apr 2 22:23:49 UTC 2021
machine[65]: x86_64
```

image

### CODE REVIEW

File: lib/posix/uname.c [hunk 1]
```diff
@@ -51,3 +51,156 @@ int uname(struct utsname *name)
 	}
 	return 0;
 }
+
+#ifdef CONFIG_POSIX_UNAME_SHELL
```
[the remark this answers is not in the document]
Comment (ycsin) [reply lib/posix/uname.c]: Good idea, thanks! Moved it to `posix/shell`

**Comment by ycsin** [comment]
Inspired by #67461 and added a commit to make the sample compilable on Linux with `make`

File: lib/posix/shell/uname.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	if (set & UNAME_UNKNOWN) {
+		shell_fprintf(sh, SHELL_NORMAL, "%s ", "unknown");
+	}
+
+	shell_fprintf(sh, SHELL_NORMAL, "\n");
+
+	return 0;
+}
+
+SHELL_CMD_REGISTER(uname, NULL, NULL, uname_cmd_handler);
```
Comment (jukkar) [inline lib/posix/shell/uname.c:171]: Just pondering here that should we have `posix` command handler and the `uname` would be a sub command for it.
If we do it like in this PR, we would have quite many commands in the "main" shell which imho clutters the shell quite a bit.

Comment (ycsin) [reply lib/posix/shell/uname.c]: I've pushed another commit for this, and chose `posix` over `sh` as the former seems to be more intuitive

**Review by jukkar — DISMISSED** [review]
LGTM

**Comment by ycsin** [comment]
image

File: lib/posix/shell/Kconfig.uname [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2024 Meta
+# SPDX-License-Identifier: Apache-2.0
+
+if POSIX_UNAME
+
+config POSIX_UNAME_SHELL
+	bool "Support for `uname` command"
+	select SHELL_GETOPT
+	select POSIX_SHELL
```
Comment (ycsin) [inline lib/posix/shell/Kconfig.uname:9]: force pushed to make it such that the top level `posix` command is compiled only if there's a sub command

**Review by jukkar — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Pull request merged**